aws-rds-dbparametergroup/src/main/java/software/amazon/rds/dbparametergroup/ListHandler.java

```diff
@@ -23,14 +23,14 @@
             final ProxyClient<RdsClient> proxyClient,
             final Logger logger) {
         setLogger(logger);
-        DescribeDbParameterGroupsResponse describeDBParameterGroupsResponse = null;
+        DescribeDbParameterGroupsResponse describeDBParameterGroupsResponse;
         DescribeDbParameterGroupsRequest describeDbParameterGroupsRequest = Translator.describeDbParameterGroupsRequest(request.getNextToken());
         try {
             describeDBParameterGroupsResponse = proxy.injectCredentialsAndInvokeV2(describeDbParameterGroupsRequest,
                     proxyClient.client()::describeDBParameterGroups);
         } catch (Exception exception) {
             return Commons.handleException(
-                    ProgressEvent.progress(request.getDesiredResourceState(), callbackContext),
+                    ProgressEvent.progress(request.getPreviousResourceState(), callbackContext),
```

Comment:
I believe this change needs an explicit comment. Looks like a mistake at first glance, even though I understand there are some reasons behind it. Would you please elaborate?

Reply:
AFAIK, In case of failure in listHandler we return last successful page retrieved. We can follow up with CFN team for further implementation details like why they expect this behaviour